capstone-ui/src/Components/ResultsBox/ResultsBox.jsx

```diff
@@ -62,7 +64,35 @@ export default function ResultsBox( { locationLevel, countyFIPS, selectedCandida
 
     candidates.sort((a, b) => b.voteCount - a.voteCount);
 
-    function handleClick(key, name) {
+    async function handleClick(key, name, e) {
```

Comment:
Ideally, we would want to prevent blocking actions for non important tasks such as logging data i.e. count for candidate clicks for given user. I think this is fine for now, but a more robust approach would be to send a request to save data at end of user session e.g. locally track candidate clicks and make a one time roundtrip request to server to store in db maybe when the user session is idle or finished.

Reply:
I'll try to do something like this. My only concern is how I'll need to combine the local clicks to the db clicks when the session is still active to build the new recommendations based on those clicks.